docs/reference/errors-and-warnings/NU1510.md

```diff
@@ -17,68 +17,82 @@ f1_keywords:
 
 ### Issue
 
-The package `System.Text.Json` has been specified for pruning through the [PrunePackageReference](../../consume-packages/Package-References-in-Project-Files.md#prunepackagereference) feature.
-The `.NET SDK` generally specifies the list of packages to be pruned. The package is not needed as a direct PackageReference since the .NET runtime itself carries either the same or higher version of the assembly.
+This warning is raised as a result of NuGet dependency graph pruning, and indicates that an otherwise prunable package was restored due to a direct `PackageReference`.
+The named package can be pruned if the direct `PackageReference` is removed, since the selected `.NET SDK` provides the same version or higher of this assembly.
 
-This warning is *only* raised when the PackageReference in question *can be completely removed from the project*.
+This only affects packages registered for pruning through the [PrunePackageReference](../../consume-packages/Package-References-in-Project-Files.md#prunepackagereference) feature.
+It is *only* raised when the `PackageReference` in question *can be completely removed from the project*.
 
 #### Example 1
+- The project's `TargetFramework` is `net10.0`.
```

Comment:
nitpick:  headings should have an empty line above and below.  I don't know what editor you used to make these changes, but in VSCode there's a markdownlint extension that gives a lot of good suggestions.